### PR TITLE
✨ Implement ExternalTask API client and its accessors

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

Initial implementation of the `ExternalTaskApiClient` and its accessors.

PR: #2

## How can others test the changes?

Use the client to access the `ExternalTaskApi` either remotely, or locally.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

**Comment by ElRaptorus** [comment]
https://github.com/process-engine/process_engine_runtime/issues/89

File: src/accessors/external_accessor.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  FinishExternalTaskRequestPayload,
+  HandleBpmnErrorRequestPayload,
+  HandleServiceErrorRequestPayload,
+  IExternalTaskApi,
+  restSettings,
+} from '@process-engine/external_task_api_contracts';
+
+export class ExternalTaskApiExternalAccessor implements IExternalTaskApi {
+
+  private baseUrl: string = 'api/external_task/v1';
```
Comment (5ebastianMeier) [inline src/accessors/external_accessor.ts]: don't want to be picky, but to not break the naming convention with the `_`-prefix this needs to be a private get property

File: src/accessors/internal_accessor.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import {IIdentity} from '@essential-projects/iam_contracts';
+
+import {
+  ExternalTask,
+  IExternalTaskApi,
+} from '@process-engine/external_task_api_contracts';
+
+export class ExternalTaskApiInternalAccessor implements IExternalTaskApi {
+
+  private readonly _externalApiService: IExternalTaskApi = undefined;
```
Comment (5ebastianMeier) [inline src/accessors/internal_accessor.ts]: What about the convention concerning get accessor and the (redundant) backing field?

File: src/external_task_api_client_service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import {
+  ExternalTask,
+  IExternalTaskApi,
+  IExternalTaskApiAccessor,
+} from '@process-engine/external_task_api_contracts';
+
+export class ExternalTaskApiClientService implements IExternalTaskApi {
+
+  private readonly _externalApiAccessor: IExternalTaskApiAccessor = undefined;
```
Comment (5ebastianMeier) [inline src/external_task_api_client_service.ts]: get accessor convention (see above)

File: src/accessors/external_accessor.ts (hunk 1 above)
Comment (roblen45) [reply src/accessors/external_accessor.ts]: Warum das?

File: src/accessors/internal_accessor.ts (hunk 2 above)
Comment (ElRaptorus) [reply src/accessors/internal_accessor.ts]: see above

File: src/external_task_api_client_service.ts (hunk 3 above)
Comment (ElRaptorus) [reply src/external_task_api_client_service.ts]: see above

**Pull request merged**